framework/src/common/errors.rs

```diff
@@ -71,3 +78,68 @@ pub enum NetBricksError {
     #[doc(hidden)]
     __Nonexhaustive,
 }
+
+#[macro_export]
+macro_rules! error_chain {
+    ($error:expr) => {
+        $crate::common::errors::print_chain($error, log::LevelFilter::Error, true)
+    };
+    () => {};
+}
+
+#[macro_export]
+macro_rules! error_causes {
+    ($error:expr) => {
+        $crate::common::errors::print_chain($error, log::LevelFilter::Error, false)
+    };
+    () => {};
+}
+
+#[macro_export]
+macro_rules! warn_chain {
+    ($error:expr) => {
+        $crate::common::errors::print_chain($error, log::LevelFilter::Warn, true)
+    };
+    () => {};
+}
+
+#[macro_export]
+macro_rules! warn_causes {
+    ($error:expr) => {
+        $crate::common::errors::print_chain($error, log::LevelFilter::Warn, false)
+    };
+    () => {};
+}
+
+/// Read a `failure` `Error` and print out the causes and a backtrace as
+/// `log::error`s
+#[doc(hidden)]
+pub fn print_chain(e: &Error, log_level: log::LevelFilter, with_backtrace: bool) {
+    if log_level == log::LevelFilter::Warn {
+        warn!("{}", e);
+        for cause in e.iter_causes() {
+            warn!("caused by: {}", cause);
+        }
+    } else if log_level == log::LevelFilter::Error {
+        error!("{}", e);
+        for cause in e.iter_causes() {
+            error!("caused by: {}", cause);
+        }
+    } else {
+        return ();
+    }
+
+    match (
+        ::std::env::var("RUST_BACKTRACE")
+            .as_ref()
+            .map(|s| s.as_str()),
+        with_backtrace,
```

Comment:
actually do we need another flag? seems to me it's enough to just key off the env variable. If you want backtrace collected and printed, set env, otherwise don't.

Reply:
@drunkirishcoder yeah. I went both ways on that. You don't think we'll want some stracktraces but not always have them "on" in like a prod-case?